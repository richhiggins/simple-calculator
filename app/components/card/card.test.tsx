import React from "react"
import { render, screen } from "@testing-library/react"

import Card from "./card"

const mockCard = (
  <Card
    linkUrl="/foo-bar"
    title="foo"
    subTitle="bar"
    price={1000}
    imageUrl="https://placekitten.com/g/200/300"
    imageAltText="foo bar"
  />
)

const mockCardNoImage = (
  <Card
    linkUrl="/foo-bar"
    title="foo"
    subTitle="bar"
    price={1000}
    imageAltText="foo bar"
  />
)

it("renders the card as expected", () => {
  const container = render(mockCard)
  expect(container).toMatchSnapshot()
})

it("renders an anchor tag as supplied", () => {
  render(mockCard)
  expect(screen.getByRole("link")).toHaveAttribute("href", "/foo-bar")
})

it("renders the card title with the correct text", () => {
  render(mockCard)
  expect(screen.getByRole("heading", { name: "foo" })).toBeInTheDocument()
})

it("renders the card sub-title with the correct text", () => {
  render(mockCard)
  expect(screen.getByRole("heading", { name: "bar" })).toBeInTheDocument()
})

it("renders the card price with the correct value", () => {
  render(mockCard)
  expect(screen.getByText("1000")).toBeInTheDocument()
})

it("renders an image when present", () => {
  render(mockCard)
  expect(screen.getByRole("img")).toBeInTheDocument()
})

it("does not render image when not present", () => {
  render(mockCardNoImage)
  expect(screen.queryByRole("img")).toBeNull()
})
