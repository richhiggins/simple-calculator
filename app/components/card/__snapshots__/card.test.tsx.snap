// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the card as expected 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="card"
        href="/foo-bar"
      >
        <img
          alt="foo bar"
          src="https://placekitten.com/g/200/300"
        />
        <h3>
          foo
        </h3>
        <h4>
          bar
        </h4>
        <div>
          1000
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="card"
      href="/foo-bar"
    >
      <img
        alt="foo bar"
        src="https://placekitten.com/g/200/300"
      />
      <h3>
        foo
      </h3>
      <h4>
        bar
      </h4>
      <div>
        1000
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
